docker/build/ionq.Dockerfile

```diff
@@ -0,0 +1,80 @@
+# ============================================================================ #
```

Comment:
We will include the tools to target IonQ by default in our packages, and hence won't need backend specific build scripts. 
Could you please remove this file so that we can proceed with this PR?

FYI, to facilitate development, we now also publish a [cuda-quantum-dev image](https://github.com/NVIDIA/cuda-quantum/pkgs/container/cuda-quantum-dev) that contains everything that is in this docker file (and more).